joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/Py2CpgOnFileSystem.scala

```diff
@@ -1,49 +1,57 @@
 package io.joern.pysrc2cpg
 
-import io.joern.x2cpg.X2Cpg
+import io.joern.x2cpg.{X2Cpg, X2CpgConfig, X2CpgFrontend}
 import io.shiftleft.codepropertygraph.Cpg
 import io.shiftleft.utils.IOUtils
 import org.slf4j.LoggerFactory
-import overflowdb.Graph
-
-import java.nio.charset.StandardCharsets
 import java.nio.file.attribute.BasicFileAttributes
 import java.nio.file.{FileVisitResult, Files, Path, Paths, SimpleFileVisitor}
 import scala.collection.mutable
+import scala.util.Try
+
+case class Py2CpgOnFileSystemConfig(
+  outputFile: Path = Paths.get(X2CpgConfig.defaultOutputPath),
+  inputDir: Path = null,
+  venvDir: Path = Paths.get(".venv"),
+  ignoreVenvDir: Boolean = true
+) extends X2CpgConfig[Py2CpgOnFileSystemConfig] {
+  override def withInputPath(inputPath: String): Py2CpgOnFileSystemConfig = {
+    copy(inputDir = Paths.get(inputPath))
+  }
 
-case class Py2CpgOnFileSystemConfig(outputFile: Path, inputDir: Path, ignoreVenvDir: Option[Path])
+  override def withOutputPath(x: String): Py2CpgOnFileSystemConfig = {
+    copy(outputFile = Paths.get(x))
+  }
+}
 
-object Py2CpgOnFileSystem {
+class Py2CpgOnFileSystem extends X2CpgFrontend[Py2CpgOnFileSystemConfig] {
   private val logger = LoggerFactory.getLogger(getClass)
 
   /** Entry point for files system based cpg generation from python code.
     * @param config
     *   Configuration for cpg generation.
     */
-  def buildCpg(config: Py2CpgOnFileSystemConfig): Cpg = {
+  override def createCpg(config: Py2CpgOnFileSystemConfig): Try[Cpg] = {
```

Comment:
Since now there is no `createCpgWithAllOverlays`, and just `createCpg`, what API should I use to apply the OSSdataflow once I get the cpg from here? Essentially wanted to do this: https://github.com/joernio/joern/pull/2124/files#diff-1bc22ed3515fa68269df0781c82f74256c1d899c8bfdaee309087a340b67cbacR51-R53 somehow via an API directly instead of